lib/grammers-client/src/client/net.rs

```diff
@@ -38,18 +39,67 @@ const DC_ADDRESSES: [(Ipv4Addr, u16); 6] = [
     (Ipv4Addr::new(91, 108, 56, 190), 443),
 ];
 
+/// WebSocket addresses to Telegram datacenters, where the index into this array
+/// represents the data center ID.
+///
+/// The addresses were obtained from the official docs.
+/// See [URI Format](https://core.telegram.org/mtproto/transports#uri-format).
+#[cfg(all(target_arch = "wasm32", target_os = "unknown"))]
+const WS_ADDRESSES: [&str; 6] = [
+    "",
+    "wss://pluto.web.telegram.org/apiws",
+    "wss://venus.web.telegram.org/apiws",
+    "wss://aurora.web.telegram.org/apiws",
+    "wss://vesta.web.telegram.org/apiws",
+    "wss://flora.web.telegram.org/apiws",
+];
+
+#[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
+pub(crate) type Transport = transport::Full;
+
+#[cfg(all(target_arch = "wasm32", target_os = "unknown"))]
+pub(crate) type Transport = transport::Obfuscated<transport::Intermediate>;
+
 const DEFAULT_DC: i32 = 2;
 
 pub(crate) async fn connect_sender(
     dc_id: i32,
     config: &Config,
-) -> Result<(Sender<transport::Full, mtp::Encrypted>, Enqueuer), AuthorizationError> {
+) -> Result<(Sender<Transport, mtp::Encrypted>, Enqueuer), AuthorizationError> {
+    #[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
     let transport = transport::Full::new();
 
-    let addr: SocketAddr = if let Some(ip) = config.params.server_addr {
-        ip
+    #[cfg(all(target_arch = "wasm32", target_os = "unknown"))]
+    let transport = transport::Obfuscated::new(transport::Intermediate::new());
+
+    let tcp_addr = DC_ADDRESSES[dc_id as usize].into();
+    let addr: ServerAddr = if let Some(ref sa) = config.params.server_addr {
+        sa.clone()
     } else {
-        DC_ADDRESSES[dc_id as usize].into()
+        #[cfg(all(not(all(target_arch = "wasm32", target_os = "unknown")),))]
+        let addr = {
+            #[cfg(not(feature = "proxy"))]
+            let addr = ServerAddr::Tcp { address: tcp_addr };
+
+            #[cfg(feature = "proxy")]
+            let addr = if let Some(proxy) = &config.params.proxy_url {
+                ServerAddr::Proxied {
+                    address: tcp_addr,
+                    proxy: proxy.to_owned(),
+                }
+            } else {
+                ServerAddr::Tcp { address: tcp_addr }
+            };
+
+            addr
+        };
+
+        #[cfg(all(target_arch = "wasm32", target_os = "unknown"))]
+        let addr = ServerAddr::Ws {
+            address: WS_ADDRESSES[dc_id as usize].to_string(),
+        };
```

Comment:
Question: can't a WS connection be proxied too?

So far, it looks like we should have something like `TcpTransport`. `WebSocketTransport` and `Proxied<T: Transport>` and `enum AnyTransport` with all the supported variants to cover all the permutations properly without creating too much mess.

Reply:
> can't a WS connection be proxied too?
No, we're in a browser. We don't have access to connection properties.

Reply:
Is the only use-case for ws is the browser? I can see it used outside of the browser too, in which case it would allow proxying at our end

Reply:
The purpose of this PR is only browser support as I don't have any use for ws in platforms other than in a browser.